lux-cli/src/lib.rs

```diff
@@ -147,6 +148,12 @@ pub enum Commands {
     /// Interact with the lux configuration.
     #[command(subcommand, arg_required_else_help = true)]
     Config(ConfigCmd),
+    /// Generate autocompletion scripts for the shell.
+    Completion {
+        /// The shell to generate the completion script for.
```

Comment:
suggestion:

- More detailed documentation.
- Make this optional and try to detect the current shell if not set
  (e.g. using `std::env::get_var("SHELL")`) - I can also do this later in a follow-up PR if you'd prefer.

```suggestion
        /// The shell to generate the completion script for.{n}
        /// If not set, Lux will try to detect the current shell.{n}
        /// Possible values: "bash", "elvish", "fish", "powershell", "zsh"{n}
```